homeassistant/components/buienradar/sensor.py

```diff
@@ -701,38 +694,22 @@ async def async_setup_entry(
 ) -> None:
     """Create the buienradar sensor."""
     config = entry.data
-    options = entry.options
-
-    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
-    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
-
-    timeframe = options.get(
-        CONF_TIMEFRAME, config.get(CONF_TIMEFRAME, DEFAULT_TIMEFRAME)
-    )
-
-    if None in (latitude, longitude):
-        _LOGGER.error("Latitude or longitude not set in Home Assistant config")
-        return
-
-    coordinates = {CONF_LATITUDE: float(latitude), CONF_LONGITUDE: float(longitude)}
+    data = entry.runtime_data
 
     _LOGGER.debug(
         "Initializing buienradar sensor coordinate %s, timeframe %s",
-        coordinates,
-        timeframe,
+        data.coordinates,
+        data.timeframe,
     )
 
     # create weather entities:
     entities = [
-        BrSensor(config.get(CONF_NAME, "Buienradar"), coordinates, description)
+        BrSensor(config.get(CONF_NAME, "Buienradar"), data.coordinates, description)
         for description in SENSOR_TYPES
     ]
 
-    # create weather data:
-    data = BrData(hass, coordinates, timeframe, entities)
-    entry.runtime_data[Platform.SENSOR] = data
-    await data.async_update()
-
+    # add entities
+    data.devices.extend(entities)
```

Comment:
What do we need this for? Ideally we don't store references to the entities anymore after adding them to hass